src/lib/generate-output.ts

```diff
@@ -16,14 +16,26 @@ interface BasicOutput {
 	filename: string | undefined;
 }
 
+/**
+ * Store a single browser instance reference so that we can re-use it.
+ */
+let browser: puppeteer.Browser | undefined;
+
+/**
+ * Close the browser instance.
+ */
+export const closeBrowser = () => browser?.close();
+
 /**
  * Generate the output (either PDF or HTML).
  */
 export async function generateOutput(html: string, relativePath: string, config: PdfConfig): Promise<PdfOutput>;
 export async function generateOutput(html: string, relativePath: string, config: HtmlConfig): Promise<HtmlOutput>;
 export async function generateOutput(html: string, relativePath: string, config: Config): Promise<Output>;
 export async function generateOutput(html: string, relativePath: string, config: Config): Promise<Output> {
-	const browser = await puppeteer.launch({ devtools: config.devtools, ...config.launch_options });
+	if (!browser) {
+		browser = await puppeteer.launch({ devtools: config.devtools, ...config.launch_options });
```

Comment:
There's a slight potential problem here that the passed config could be different per source file, but both `devtools` and `launch_options` can not be set per file at the moment, iirc, so it shouldn't be a problem for now.

Reply:
I did not look yet more into the config possibilities, so I can't comment this issue.

But maybe it would be possible to make the config arguments optional and then use the "default" browser or create a new one depending on these arguments.

Could this whole approach with the browser better solved as a Class?

Reply:
I think it's fine as is for now but yeah potentially would need to check the config and create additional browser instances for each unique config. Since both `devtools` and `launch_options` can only be set globally (not via front-matter per file), right now there's no issue here.